x-pack/legacy/plugins/maps/public/layers/styles/vector/components/symbol/icon_stops.test.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getFirstUnusedSymbol } from './icon_stops';
+
+describe('getFirstUnusedSymbol', () => {
+  const symbolOptions = [{ value: 'icon1' }, { value: 'icon2' }];
+
+  test('Should return first unused icon from PREFERRED_ICONS', () => {
+    const iconStops = [
+      { stop: 'category1', icon: 'circle' },
+      { stop: 'category2', icon: 'marker' },
+    ];
+    const nextIcon = getFirstUnusedSymbol(symbolOptions, iconStops);
+    expect(nextIcon).toBe('square');
+  });
+
+  test('Should fallback to first unused general icons when all PREFERRED_ICONS are used', () => {
+    const iconStops = [
+      { stop: 'category1', icon: 'circle' },
+      { stop: 'category2', icon: 'marker' },
```

Comment:
these duplicate `category2` values would fail to validate if the app would actually run. Both due to the UX-validation, but also in the downstream mapbox rule.

maybe replace with valid example. e.g.:

```
const iconStops = ["circle", "marker", "square", "star", "triangle", "hospital", "circle-stroked", "marker-stroked", "square-stroked", "star-stroked", "triangle-stroked", "icon1"].map((preferredIcon, i) => {
    return {stop: `category${i}`, icon: preferredIcon};
});
```